app/models/piece.rb

```diff
@@ -100,4 +100,22 @@ def is_obstructed_diagonally(piece_destination)
       end 
     end 
   end 
+
+  def move_to!(x_new, y_new)
+    x_current = piece.position_x
+    y_current = piece.position_y
+    x_destination = position_x
+    y_destination = position_y
+
+    #moving to an empty space, move is valid
+    if ! is_occupied?(x_destination, y_destination) && valid_move?(x_destination, y_destination)
+      piece.update_attributes(:position_x => x_destination, :position_y => y_destination)
+    #moving to an occupied space, move is valid
+    #the valid_move? method covers the color of the piece
+    elsif is_occupied?(x_destination, y_destination) && valid_move?(x_destination, y_destination)
+      piece[x_destination, y_destination].delete
```

Comment:
Few things here: 
1. Does this actually delete the piece at the destination?
2. What if this is my own piece?

Reply:
@ilyakrasnov the valid_move? method (at least for the king, so far) covers whether or not a piece is ones own piece, so it was not necessary to repeat in this method.

Reply:
@snsmith92 got it, thats totally ok for now.. once the other methods are done we can refactor.. it.

@drewsef-916 what does `piece[x_destination,  y_destination]` do? sure that it is the right syntax?

Reply:
@ilya piece[x_destination, y_destination] should call on the piece with the coordinates that self wants to move to (i.e. the destination). Indeed our syntax here is probably not correct... Perhaps changed to `Piece.where(position_y = y_destination and position_x = x_destination)` something of that sort...?

Reply:
Exactly! Using `Piece.where..` you will need to add `game_id` as well.. otherwise you may get a piece from another game.. you could though also use `game.pieces.where` which will automatically scope to only this game's pieces

Reply:
Got it! 👍